crates/bevy_asset/src/loader.rs

```diff
@@ -146,20 +143,20 @@ impl<'a> LoadContext<'a> {
 
 /// The result of loading an asset of type `T`
 #[derive(Debug)]
-pub struct AssetResult<T: Component> {
+pub struct AssetResult<T> {
```

Comment:
This surprises me a bit: why were we able to remove this component bound / why did it exist in the first place?

Reply:
I'm not sure exactly. I was surprised about that bound as well, as it didn't make much sense to me. I think it bound should be more like `T: Asset`, but clearly that bound wasn't used either. Removing it seems to be just fine, so I went with that.